### ISSUE TITLE
MKS Plugin for Cura 4.4 ?

### ISSUE THREAD

**Issue by vicodin77** [makerbase-mks/Software#14]
Would be great to become a MKS Plugin for the new Version of Cura ..

Greetings Mindreader

**Comment by Jeredian** [comment]
I'm not sure how often developers from makerbase make new releases, so I copied the last version of the plugin and fixed the issue for the latest version of Cura.
Here is a link to my repository: https://github.com/Jeredian/mks-wifi-plugin
If there will be an official repository with source codes of the plugin, I'll be glad to make a pull request into it.

**Comment by Jeredian** [comment]
At now the plugin is available through the Cura Marketplace as the MKS Wifi Plugin. And it supports Cura versions from 3.5.0+ to 4.4.0+. 
I linked in description to the official repository, and the latest version is from my fork.
I hope it will help to use the plugin more comfortable.

**Comment by MarshMalloW0007** [comment]
Save as TFT file does'nt work :

2019-12-21 19:48:10,166 - DEBUG - [MainThread] MKSWifiPlugin.MachineConfig._createAdditionalComponentsView [149]: Creating additional ui components for tft35.
2019-12-21 19:48:10,245 - DEBUG - [MainThread] MKSWifiPlugin.MachineConfig.startDiscovery [45]: Starting printer discovery.
2019-12-21 19:48:10,370 - CRITICAL - [MainThread] cura.CrashHandler.__init__ [61]: An uncaught error has occurred!
2019-12-21 19:48:10,376 - CRITICAL - [MainThread] cura.CrashHandler.__init__ [64]: Traceback (most recent call last):
2019-12-21 19:48:10,383 - CRITICAL - [MainThread] cura.CrashHandler.__init__ [64]:   File "C:\Users\***\AppData\Roaming\cura\4.4\plugins\MKSWifiPlugin\MKSWifiPlugin\MachineConfig.py", line 48, in startDiscovery
2019-12-21 19:48:10,388 - CRITICAL - [MainThread] cura.CrashHandler.__init__ [64]:     self._network_plugin.printerListChanged.connect(self._onPrinterDiscoveryChanged)
2019-12-21 19:48:10,393 - CRITICAL - [MainThread] cura.CrashHandler.__init__ [64]: AttributeError: 'NoneType' object has no attribute 'printerListChanged'

**Comment by Jeredian** [comment]
Can you it from latest master branch from my repository?(https://github.com/Jeredian/mks-wifi-plugin) If this issue still exists please add an issue inside my repository,and I will check it as soon I can

**Comment by MarshMalloW0007** [comment]
Well, that's good news, 
It's working.
Thx for this quick response

**Comment by Phacops-rana** [comment]
Hello, I have a little bit silly problem with MKS plugin. Im on Cura 4.4.1 and Jeredian`s version of MKS plugin, printer is Two trees Sapphire Pro. Thing is, that while the WiFi communication works, the previews of gcodes are all corrupted (see pic) [https://imgur.com/bErm1yD](url)
The best part is, that I managed to get working previews with older version of plugin, but it corrupts names of gcode files sent to the printer (they are all changed to cura-file.gcode)!
How can I get both preview and name of the file?

**Comment by Jeredian** [comment]
Hi, @Phacops-rana . 
You can look on this issue https://github.com/Jeredian/mks-wifi-plugin/issues/11
It's related to this behavior and will be fixed in the next plugin update

**Comment by Phacops-rana** [comment]
Oh, thank you @Jeredian for quick update! I will try this solution

**Comment by Phacops-rana** [comment]
Thank you very much, I am new to GitHub and it is quite complicate to
navigate, so I dublicated my message

пн, 13 янв. 2020, 21:29 Pavel Polyakov <notifications@github.com>:

**Comment by marcohaddad** [comment]
Hello,

I solved the "corruption"  problem.

The preview image was being created with the wrong size for my MKS-TFT35. It was 50 by 50 pixels instead of 100 by 100 pixels.

I just edited the SaveOutputDevice.py file and replaced on line 120:
save_file.write(utils.add_screenshot(image, 50, 50, ";simage:"))
with:
save_file.write(utils.add_screenshot(image, 100, 100, ";simage:"))

Best regards,

**Comment by bilou24an199** [comment]
same things than marcohaddad

i modified the lines indicated and it is working properly

BR.

**Comment by Phacops-rana** [comment]
Finally, I dialed it in: for Two Trees Sapphire Pro correct numbers are simage 100, gimage 200

**Comment by Zdendo463** [comment]
Yes, simage resolution fixing problem. Now after some update there is color depth problem (seems like CGA graphic). Is there anyone, who knows how to fix this? Thx.

**Comment by Jeredian** [comment]
Hi @Zdendo463  , you can add a new issue into mks plugin repository (https://github.com/makerbase-mks/mks-wifi-plugin)  or into my repository  (https://github.com/Jeredian/mks-wifi-plugin).
